lib/TH/vector/AVX.c

```diff
@@ -0,0 +1,243 @@
+#ifndef _MSC_VER
+#include <x86intrin.h>
+#else
+#include <intrin.h>
+#endif
+
+static void THDoubleVector_fill_AVX(double *x, const double c, const ptrdiff_t n) {
+  ptrdiff_t i;
+  ptrdiff_t off;
+  __m256d YMM0 = _mm256_set_pd(c, c, c, c);
+  for (i=0; i<=((n)-16); i+=16) {
+    _mm256_storeu_pd((x)+i  , YMM0);
```

Comment:
The compiler doesn't do loop unrolling for you? I'm surprised you're seeing a benefit from manually unrolling these. Any idea why?